### PR TITLE
RHIROS-600  Executive report

### PR DESCRIPTION
**Pull request by PreetiW**
### Description

Jira: https://issues.redhat.com/browse/RHIROS-600

### Demo

https://drive.google.com/file/d/18C9-L8u4E2GTl6T4LgShS6cwgPHxXRmu/view?usp=sharing

### CODE REVIEW

File: src/Components/Reports/Common/ExecutiveSecondPage.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+export const ExecutiveSecondPage = () => {
+
+    return <Fragment key="second-page">
+        <Text style={styles.execHeading}>Description of states</Text>
+        <Dl>
+            <Dt>
+                <View style={{ display: 'flex',
+                    flexDirection: 'row' }}>
+                    <Canvas
```
Comment (r14chandra) [inline src/Components/Reports/Common/ExecutiveSecondPage.js:19]: `CanvasComponent` can be used here as well. (we can use different name though 😅  )

Comment (r14chandra) [inline src/Components/Reports/Common/ExecutiveSecondPage.js:19]: `CanvasComponent` can be used here as well. (we can use different name though 😅  )

**Review by r14chandra — COMMENTED** [review]
Let me know your thoughts on modularizing the code.

File: src/Components/Reports/Common/ExecutiveFirstPage.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                    style={{
+                        width: 15,
+                        height: 10,
+                        padding: 2
+                    }}
+                    paint={({ path, scale }) => {
+                        scale(0.014);
+                        path(CircleIconConfig.svgPath).fill('#8BC1F7');
+                    }}
+                />
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/Reports/Common/ExecutiveFirstPage.js]: Thank you @r14chandra for suggestion, I have extracted the logic and created common components, have pushed the code. Let me know if you have any other suggestions.

File: src/Components/Reports/Common/ExecutiveSecondPage.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        </Dl>
+
+        <Text style={styles.execHeading}>Description of conditions</Text>
+        <Dl>
+            <Dt>
+                <View style={{ display: 'flex',
+                    flexDirection: 'row' }}>
+                    <Text>CPU pressure</Text>
+                </View>
+            </Dt>
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/Reports/Common/ExecutiveSecondPage.js]: Thank you @r14chandra for suggestion, I have extracted the logic and created common components. Let me know if you have any other suggestions.

File: src/Components/Reports/Common/ExecutiveSecondPage.js (hunk 1 above)
Comment (PreetiW) [reply src/Components/Reports/Common/ExecutiveSecondPage.js]: Thank you @r14chandra for suggestion, I have extracted the logic and created common components. Let me know if you have any other suggestions.

**Review by r14chandra — APPROVED** [review]
Looks good to me now! 👍🏻 Thanks @PreetiW!

File: src/Components/Reports/Util.js [hunk 4]
```diff
@@ -70,3 +70,68 @@ export const getSystemsReportFileName = () =>  {
...
+
+        conditionsChartData.push({ x: conditionName, y: percentage });
+        conditionsTableData.push([`${count}`]);
+
+        const underPressureValue = conditions[condition].under_pressure ? conditions[condition].under_pressure : 0;
+        const undersizedValue = conditions[condition].undersized ? conditions[condition].undersized : 0;
+        const oversizedValue = conditions[condition].oversized ? conditions[condition].oversized : 0;
+
+        condtionsInfo[condition].occurances.push(['Under pressure', `${underPressureValue}`]);
+        if (condition !== 'io') {
```
Comment (kgaikwad) [inline src/Components/Reports/Util.js:130]:
```suggestion
        if (condition === 'io' && undersizedValue != -1 && oversizedValue != -1) {
```

Using this condition, I think we won't need to touch frontend once backend is ready for IOPS. What do you think?

Comment (PreetiW) [reply src/Components/Reports/Util.js:130]: good point @kgaikwad I have made the above change and will push it in the latest commit 💯

**Review by kgaikwad — CHANGES_REQUESTED** [review]
@PreetiW  - As to get PSI enabled systems info will take some time can we include line of information under tables which applies to every situation?

`Under pressure conditions can be encountered when systems are having PSI enabled`

**Comment by PreetiW** [comment]
> @PreetiW - As to get PSI enabled systems info will take some time can we include line of information under tables which applies to every situation?
> 
> `Under pressure conditions can be encountered when systems are having PSI enabled`

@kgaikwad  have added the message and pushed the changes 🚀

**Review by kgaikwad — APPROVED** [review]

**Pull request merged**